docs/docs/behaviors/macros.md

```diff
@@ -0,0 +1,136 @@
+once the macro is released.
+
+## Macro Definition
+
+Each macro you want to use in your keymap gets defined first, then bound in your keymap.
+
+A macro definition looks like:
+
+```
+/ {
+    macros {
+        zed_em_kay: zmk_macro {
+        label = "ZMK_MAC";
+        compatible = "zmk,behavior-macro";
+        #binding-cells = <0>;
+        bindings
+            = <&macro_press &kp LSHFT>
+            , <&macro_tap &kp Z &kp M &kp K>
+            , <&macro_release &kp LSHFT>
+            ;
+        };
+    };
+};
+```
+
+:::note
+The text before the colon (`:`) in the declaration of the macro node is the "node label", and is the text
+used to reference the macro in your keymap
+:::
+
+The macro can then be bound in your keymap by referencing it by the label `&zed_em_kay`, e.g.:
+
+```
+        raise_layer {
+            bindings = <&zed_em_kay>;
+        };
+```
+
+### Bindings
+
+Like [hold-taps](/docs/behaviors/hold-tap), macros are created by composing other behaviors, and any of those behaviors can
+be added to the `bindings` list, e.g.:
+
+```
+bindings
+    = <&to 1>
+    , <&bl BL_ON>
+    , <&kp Z &kp M &kp K &kp EXCLAMATION>
+    ;
+```
+
+## Macro Controls
+
+There are a set of special macro controls that can be included in the `bindings` list to modify the
+way the macro is processed.
+
+### Binding Activation Mode
+
+Bindings in a macro are activated differently, depending on the current "activation mode" of the macro.
+
+Available modes:
+
+- Tap - The default mode; when in this mode, the macro will press, then release, each behavior in the `bindings` list. This mode is useful for
+  basic keycode output to hosts, i.e. when activating a `&kp` behavior.
+- Press - In this mode, the macro processing will only trigger a press on the behaviors in the macro. This is useful for holding down modifiers for some duration of a macro, e.g. `&kp LALT`.
+- Release - In this mode, the macro processing will only trigger a release on the behaviors in the macro. This is useful for releasing modifiers previously pressed earlier in the macro processing, e.g. `&kp LALT`.
+
+To modify the activation mode, macro controls can be added at any point in the bindings list.
+
+- `&macro_tap`
+- `&macro_press`
+- `&macro_release`
+
+A concrete example, used to hold a modifier, tap multiple keys, then release the modifier, would look like:
+
+```
+bindings
+    = <&macro_press &kp LSHFT>
+    , <&macro_tap &kp Z &kp M &kp K>
+    , <&macro_release &kp LSHFT>
+    ;
+```
+
+### Processing Continuation on Release
+
+The macro can be paused so that only part of the `bindings` list is processed when the macro is pressed, and the remainder is processed once
+the macro itself is released.
+
+To pause the macro until release, use `&macro_pause_for_release`, like in this example that will press a modifier and activate a layer when the macro is pressed, and once the macro is released, release the modifier and deactivate the layer by releasing the `&mo`:
```

Comment:
Feels a little bit like a run on sentence. Maybe split this up?
```suggestion
To pause the macro until release, use `&macro_pause_for_release`. For example, this macro will press a modifier and activate a layer when the macro is pressed. Once the macro is released, it will release the modifier and deactivate the layer by releasing the `&mo`:
```